Automattic/jetpack
# Add additional properties to WooCommerce analytics checkout and purchase events

Fixes https://github.com/woocommerce/woocommerce-blocks/issues/9729

## Proposed changes:

- Adds several new properties to the WooCommerce Analytics events:
  - To the `woocommerceanalytics_product_checkout` event:
	`shipping_option`: the selected shipping option 
	`products_count`: number of products included in the cart
	`coupon_used`: if the coupon was used (boolean)
	`order_value`: the order value/total
	`template_used`: if the proceed from cart to checkout action has happened on a regular cart page or powered by the template
	`additional_blocks`: if there are additional blocks on the cart page
	`store_currency`: the currency used in the store

  - To the `woocommerceanalytics_product_purchase` event
		`products_count`: number of products included in the cart
		`coupon_used`: if the coupon was used (boolean)
		`order_value`: the order value/total
		`template_used`: if the proceed from cart to checkout action has happened on a regular cart page or powered by the template
		`additional_blocks`: if there are additional blocks on the cart page
		`store_currency`: the currency used in the store

### Other information:


## Jetpack product discussion


A previous discussion about this tracking can be found here pb0Spc-ou-p2#comment-701 however it is very old.

We also have pca54o-6dR-p2 where the effort is listed as a cycle goal (point 3).

## Does this pull request change what data or activity we track or use?
Yes, it adds the above mentioned properties to the data tracked when viewing the checkout page and when purchasing a product on WooCommerce.

## Testing instructions:





- Ensure your store is set up so it can be tracked.
- Ensure your site is using the Checkout block.
- Set up multiple shipping methods on your store.
- Install [Multiple Packages for WooCommerce](https://wordpress.org/plugins/multiple-packages-for-woocommerce/). Go to the settings (WooCommerce -> Settings -> Multiple Packages) and set the grouping option to "Product (individual)".
- Install the Tracks Vigilante extension (optional)
- For each of the following scenarios, view the Checkout page also complete the checkout process, ensuring the `woocommerceanalytics_product_checkout` and `woocommerceanalytics_product_purchase` events fire with the correct properties (Except `shipping_option` on the purchase event)
- Add multiple items to your cart (including a virtual item), go to the Checkout page and change the shipping option on each of them.
- Check out and check the events are correct and the correct shipping option is tracked, ensure a shipping option is not tracked for the virtual item.
- Do this again, this time check out with a coupon.
- Go to Appearance -> Editor and edit the Checkout template. Add some extra blocks to the template.
- Go to the Checkout page again and ensure the extra blocks you added are in the `additional_blocks` property.
- Downgrade your WooCommerce version to 7.8.2 and ensure WooCommerce Blocks is disabled.
- Add products to your cart and check out. Ensure the `template_used` value is false (Cart/Checkout templates were not in use in this version).
- Go to the Checkout page (Pages -> Checkout) and add extra blocks to the Checkout page.
- Go to the Checkout block and ensure the additional blocks you added are in the `additional_blocks` property.
- Change the checkout page to use the shortcode (`[woocommerce_checkout]`) and repeat these steps. Ensure the correct value is tracked for each event. **Note that the `checkout_page_contains_cart_shortcode` property is stored in a transient so this will be incorrect if changing, but it has been tested in another PR and is known to be working.**
